integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/MediationWithNashornJsScriptStoredInRegistryTestCase.java

```diff
@@ -0,0 +1,109 @@
+            + "invoking it with the given 'key'")
+    public void testJSMediatorWithTheGivenKey() throws Exception {
+
+        OMElement response = axis2Client.sendCustomQuoteRequest(
+                getProxyServiceURLHttp("scriptMediatorNashornJsRetrieveFromRegistryTestProxy"), null,
+                "NashornInvokeFromRegistry");
+
+        assertNotNull(response, "Fault response message null");
+
+        assertNotNull(response.getQName().getLocalPart(), "Fault response null localpart");
+        assertEquals(response.getQName().getLocalPart(), "CheckPriceResponse", "Fault localpart mismatched");
+
+        assertNotNull(response.getFirstElement().getQName().getLocalPart(), " Fault response null localpart");
+        assertEquals(response.getFirstElement().getQName().getLocalPart(), "Code", "Fault localpart mismatched");
+        assertEquals(response.getFirstElement().getText(), "NashornInvokeFromRegistry", "Fault value mismatched");
+
+
+        assertNotNull(response.getFirstChildWithName(new QName("http://services.samples/xsd", "Price")), "Fault response null localpart");
+
+
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        super.cleanup();
+        clearUploadedResource();
+    }
+
+
+    private void uploadResourcesToConfigRegistry() throws Exception {
+
+        ResourceAdminServiceClient resourceAdminServiceStub =
+                new ResourceAdminServiceClient(contextUrls.getBackEndUrl(), context.getContextTenant().getContextUser().getUserName()
+, context.getContextTenant().getContextUser().getPassword());
+
+        resourceAdminServiceStub.deleteResource("/_system/config/script_js");
+        resourceAdminServiceStub.addCollection("/_system/config/", "script_js", "", "Contains test js files");
+
+        resourceAdminServiceStub.addResource("/_system/config/script_js/stockquoteTransformNashorn.js",
+                "application/x-javascript", "js files", new DataHandler(new URL("file:///" + getESBResourceLocation()
+                        + "/mediatorconfig/script_js/stockquoteTransformNashorn.js")));
+    }
+
+    private void enableDebugLogging() throws Exception {
+        LoggingAdminClient logAdminClient = new LoggingAdminClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logAdminClient.updateLoggerData("org.apache.synapse", "DEBUG", true, false);
+    }
+
+    private void clearUploadedResource()
+            throws InterruptedException, ResourceAdminServiceExceptionException, RemoteException, XPathExpressionException {
+
+        ResourceAdminServiceClient resourceAdminServiceStub =
+                new ResourceAdminServiceClient(contextUrls.getBackEndUrl(), context.getContextTenant().getContextUser().getUserName()
+, context.getContextTenant().getContextUser().getPassword());
+
+        resourceAdminServiceStub.deleteResource("/_system/config/script_js");
+        Thread.sleep(1000);
```

Comment:
Can we improve this sleep to wait until a condition? ( while resource does not exist etc)

Reply:
+1, Also let's move that code to a common place so that is can be used in all such places.